crux_time/src/lib.rs

```diff
@@ -44,33 +61,135 @@ where
         Self { context }
     }
 
-    /// Request current time, which will be passed to the app as `chrono::DateTime<Utc>`
+    /// Request current time, which will be passed to the app as a [`TimeResponse`] containing an [`Instant`]
     /// wrapped in the event produced by the `callback`.
     pub fn now<F>(&self, callback: F)
     where
-        F: Fn(DateTime<Utc>) -> Ev + Send + Sync + 'static,
+        F: Fn(TimeResponse) -> Ev + Send + Sync + 'static,
     {
         self.context.spawn({
             let context = self.context.clone();
             async move {
-                let response = context.request_from_shell(TimeRequest).await;
-                let response = response
-                    .0
-                    .parse::<DateTime<Utc>>()
-                    .expect("invalid time format");
-                context.update_app(callback(response));
+                context.update_app(callback(context.request_from_shell(TimeRequest::Now).await));
             }
         });
     }
 
-    /// Request current time, which will be returned as `chrono::DateTime<Utc>`.
+    /// Request current time, which will be passed to the app as a [`TimeResponse`] containing an [`Instant`]
     /// This is an async call to use with [`crux_core::compose::Compose`].
-    pub async fn now_async(&self) -> DateTime<Utc> {
+    pub async fn now_async(&self) -> TimeResponse {
+        self.context.request_from_shell(TimeRequest::Now).await
+    }
+
+    /// Subscribe to receive a notification when the specified [`Instant`] has arrived.
+    pub fn subscribe_instant<F>(&self, instant: Instant, callback: F)
+    where
+        F: Fn(TimeResponse) -> Ev + Send + Sync + 'static,
+    {
+        self.context.spawn({
+            let context = self.context.clone();
+            async move {
+                context.update_app(callback(
+                    context
+                        .request_from_shell(TimeRequest::SubscribeInstant(instant))
+                        .await,
+                ));
+            }
+        });
+    }
+
+    /// Subscribe to receive a notification when the specified [`Instant`] has arrived.
+    /// This is an async call to use with [`crux_core::compose::Compose`].
+    pub async fn subscribe_instant_async(&self, instant: Instant) -> TimeResponse {
         self.context
-            .request_from_shell(TimeRequest)
+            .request_from_shell(TimeRequest::SubscribeInstant(instant))
             .await
-            .0
-            .parse::<DateTime<Utc>>()
-            .expect("invalid time format")
+    }
```

Comment:
If you're into the `sleep` suggestion I made elsewhere, maybe this is `sleep_until`?